docs/development.md

```diff
@@ -0,0 +1,94 @@
+# Development Guide
+
+## Workflow
+
+### Fork the main repository
+
+1. Go to https://github.com/redhat-developer/odo
+2. Click the "Fork" button (at the top right)
+
+### Clone your fork
+
+The commands below require that you have $GOPATH. We highly recommended you put odo code into your $GOPATH.
+
+```sh
+git clone https://github.com/$YOUR_GITHUB_USERNAME/odo.git $GOPATH/src/github.com/redhat-developer/odo
+cd $GOPATH/src/github.com/redhat-developer/odo
+git remote add upstream 'https://github.com/redhat-developer/odo'
+```
+
+### Create a branch and make changes
+
+```sh
+git checkout -b myfeature
+# Make your code changes
+```
+
+### Keeping your development fork in sync
+
+```sh
+git fetch upstream
+git rebase upstream/master
+```
+
+Note: If you have write access to the main repository at github.com/redhat-developer/odo, you should modify your git configuration so that you can't accidentally push to upstream:
+
+```sh
+git remote set-url --push upstream no_push
+```
+
+### Committing changes to your fork
+
+```sh
+git commit
+git push -f origin myfeature
+```
+
+### Creating a pull request
+
+1. Visit https://github.com/$YOUR_GITHUB_USERNAME/odo.git
+2. Click the "Compare and pull request" button next to your "myfeature" branch.
+3. Check out the pull request process for more details
+
+## Dependency Management
+
+odo uses `glide` to manage dependencies and `glide-vc` to clean vendor directory.
+
+They are not strictly required for building odo but they are required when managing dependencies under the `vendor/` directory.
+
+If you want to make changes to dependencies please make sure that `glide` and `glide-vc` are installed and are in your `$PATH`.
```

Comment:
could you also add a line saying -
to modify the dependencies, make changes in glide.yaml file, and then run `glide update -v`
something like that?